internal/config/validator.go

```diff
@@ -58,13 +58,65 @@ func (v *validator) Validate(resources ...client.ObjectList) error {
 			clusterResources.Nodes = append(clusterResources.Nodes, list.Items...)
 		}
 	}
+	clusterResources = resetTransientErrorsFields(clusterResources)
 	_, err := For(clusterResources, v.validate)
-	if errors.As(err, &TransientError{}) { // we do not want to make assumption on ordering in webhooks.
-		return nil
-	}
 	return err
 }
 
 func NewValidator(validate Validate) apivalidate.ClusterObjects {
 	return &validator{validate: validate}
 }
+
+// resetting the fields that lead to Transienterror.
+func resetTransientErrorsFields(clusterResources ClusterResources) ClusterResources {
+	for i := range clusterResources.Peers {
+		clusterResources.Peers[i].Spec.BFDProfile = ""
+		clusterResources.Peers[i].Spec.PasswordSecret.Name = ""
+	}
+	for i, bgpAdv := range clusterResources.BGPAdvs {
+		var communities []string
+		for _, community := range bgpAdv.Spec.Communities {
+			if existCommunity(community, clusterResources.Communities) {
+				communities = append(communities, community)
+			} else if !checkInvalidCommunityFormat(community) {
+				communities = append(communities, community)
+			}
+		}
+		clusterResources.BGPAdvs[i].Spec.Communities = communities
+		bgpAdv.Spec.Communities = communities
```

Comment:
why is this needed?

Reply:
Because I save the entire communities that don't generate a Transienterror, and when I finish the loop that occurs in lines 78-84, I update the 'communities' field. 
The goal is to update the field and ensure that the entire set of communities doesn't create a Transienterror.

Reply:
I'm talking specifically about line 86, why isn't
```go
clusterResources.BGPAdvs[i].Spec.Communities = communities
```
enough?

Reply:
Oh! you're right, tnx!